src/app.js

```diff
@@ -373,75 +194,53 @@ var run = function() {
 	$filesEl.on('click', '.file .remove', function(ev) {
 		ev.preventDefault();
 		var name = $(this).parent().find('.name').text();
-		var index = getFiles().indexOf( fileKey(name) );
 
 		if (confirm("Are you sure you want to remove: " + name + " from local storage?")) {
-			window.localStorage.removeItem( fileKey( name ) );
-			SOL_CACHE_FILE = getFiles()[ Math.max(0, index - 1)];
+			window.localStorage.removeItem( utils.fileKey( name ) );
+			editor.setNextFile(utils.fileKey(name));
```

Comment:
No, because it's using the name to find the *next* file. (Since the named file has been removed.) There are a flurry of subsequent minor refactorings to do once we get this major one out of the way.